app/main.py

```diff
@@ -1 +1,20 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import VaccineError, NotWearingMaskError, NotVaccinatedError
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    masks_needed = 0
+
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except NotVaccinatedError:
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            masks_needed += 1
+        except VaccineError:
+            return "All friends should be vaccinated"
```

Comment:
The `VaccineError` exception is a base class for `NotVaccinatedError` and `OutdatedVaccineError`. Since `NotVaccinatedError` is already caught, catching `VaccineError` separately is redundant. Consider removing this catch block or handling `OutdatedVaccineError` specifically if needed.